src/containers/pages/IRS/Map/index.tsx

```diff
@@ -0,0 +1,336 @@
+  structures: null,
+};
+
+/** The IRS Reporting Map component */
+const IRSReportingMap = (props: IRSReportingMapProps & RouteComponentProps<RouteParams>) => {
+  const [loading, setLoading] = useState<boolean>(true);
+  const {
+    fetchFocusAreas,
+    fetchJurisdictionsAction,
+    fetchPlans,
+    focusArea,
+    fetchStructures,
+    jurisdiction,
+    plan,
+    service,
+    structures,
+  } = props;
+
+  let planId: string | null = null;
+  if (props.match && props.match.params && props.match.params.planId) {
+    planId = props.match.params.planId;
+  }
+
+  let jurisdictionId: string | null = null;
+  if (props.match && props.match.params && props.match.params.jurisdictionId) {
+    jurisdictionId = props.match.params.jurisdictionId;
+  }
+
+  /** async function to load the data */
+  async function loadData() {
+    try {
+      setLoading(!focusArea || !jurisdiction || !plan || !structures); // set loading when there is no data
+
+      let fetchLocationParams: SupersetFormData | null = null;
+      if (jurisdictionId) {
+        fetchLocationParams = superset.getFormData(1, [
+          { comparator: jurisdictionId, operator: '==', subject: 'jurisdiction_id' },
+        ]);
+      }
+
+      // get the jurisdiction
+      await service(SUPERSET_JURISDICTIONS_SLICE, fetchLocationParams).then(
+        (result: Jurisdiction[]) => fetchJurisdictionsAction(result)
+      );
+
+      let fetchStructureParams: SupersetFormData | null = null;
+      if (jurisdictionId) {
+        fetchStructureParams = superset.getFormData(3000, [
+          { comparator: jurisdictionId, operator: '==', subject: 'jurisdiction_id' },
+        ]);
+      }
+
+      // get the structures
+      await service(SUPERSET_IRS_REPORTING_STRUCTURES_DATA_SLICE, fetchStructureParams).then(
+        (result: GenericStructure[]) =>
+          fetchStructures(SUPERSET_IRS_REPORTING_STRUCTURES_DATA_SLICE, result)
+      );
+
+      if (focusAreaSlice) {
+        let fetchFocusAreaParams: SupersetFormData | null = null;
+        if (jurisdictionId && planId) {
+          fetchFocusAreaParams = superset.getFormData(1, [
+            { comparator: jurisdictionId, operator: '==', subject: 'jurisdiction_id' },
+            { comparator: planId, operator: '==', subject: 'plan_id' },
+          ]);
+        }
+        // get the focus area
+        await service(focusAreaSlice, fetchFocusAreaParams).then((result: GenericJurisdiction[]) =>
+          fetchFocusAreas(focusAreaSlice, result)
+        );
+      }
+
+      let fetchPlansParams: SupersetFormData | null = null;
+      if (planId) {
+        fetchPlansParams = superset.getFormData(1, [
+          { comparator: planId, operator: '==', subject: 'plan_id' },
+        ]);
+      }
+
+      // get the plan
+      await service(SUPERSET_IRS_REPORTING_PLANS_SLICE, fetchPlansParams).then(
+        (result: IRSPlan[]) => fetchPlans(result)
+      );
+    } catch (e) {
+      // do something with the error?
```

Comment:
what about creating an issue for this? notifying the user or retry seem like things we could do, but issue can just refer to this and say we need to discuss, decide, then implement

Reply:
this can also be a `TODO: clean up in [pasted link to issue]`

Reply:
Ultimately I think this requires a few design decisions on how we handle errors in general, a while back we had talked about Growl-like notifications (which would include but not be limited to error handling), but this needs scoping.

https://github.com/onaio/reveal-frontend/issues/300